### PR TITLE
Updating trigger node properties

### PR DESCRIPTION
**Pull request by armaneous**

### CODE REVIEW

File: stack-audit-cloud-threats/services/config.rb [hunk 1]
```diff
@@ -89,10 +89,13 @@
   QUERY
   meta_rule_node_triggers ({
     'internet_gateway' => ['relates_to'], 
-    'route' => [], 'route_table' => [], 
+    'route' => [], 
+    'route_table' => [], 
     'route_table_association' => [], 
     'iam_instance_profile' => [], 
-    'role' => []
```
Comment (smattes) [inline stack-audit-cloud-threats/services/config.rb]: I'm just curiuos - what is the effect of a missing comma here?

Comment (armaneous) [reply stack-audit-cloud-threats/services/config.rb]: Probably an engine compilation error. It's a Ruby Hash, the commas delimit the key-value pair, so the error would be:
```
syntax error, unexpected tIDENTIFIER, expecting '}'
```

**Comment by armaneous** [comment]
@scvenema Sequence of events:

- Private instance => api return doesn't include `private_ip_address`
- Instance made public => api return includes `private_ip_address`
- Instance made private => api return doesn't include `private_ip_address`

If we trigger on `private_ip_address`, we won't detect when the issue has been remediated. We could force, in the adapter, that `private_ip_address` always exists and defaults to `""` (empty string). That'd allow us to trigger on that predicate.

**Comment by armaneous** [comment]
Another issue is that we also don't know _where_ to interpolate the changed uids. `<%= filter['instance'] %>` is supposed to be switched out with uids of changed nodes when triggered on `instance` types, but because we do it twice we don't know where to interpolate. Ideally, we'd run the query twice and interpolate one without the other, then again with that flipped and another query to get the intersection of the two. This actually is a fairly complicated query with our current schema.

**Comment by scvenema** [comment]
I'm not clear why you are referring to *private* ip address fields in above comment? There is a `PublicIpAddress` field returned by `describe-instances()` which is only present if the instance has a public IP address. Seems like this could (should?) be used?

Hmm...so if property disappears, we presumably still get a CWE and then we RE should get a delta from Rosetta when the graph is updated? Or are we not updating the associated property in the 'instance' nodes?

**Comment by armaneous** [comment]
@scvenema Sorry, I don't know how that happened and I just got stuck on _private_ instead of _public_. I've updated the comment to reflect that.

If the property disappears, griffin won't register that there's a delta on the property since it won't see to insert an overriding value.

**Comment by scvenema** [comment]
Well if this wasn't metaprogramming, I'd just insert a conditional logic clause that if the `PublicIpAddress` wasn't present then it would set the value to 'nil' or some other signal like that which would change the graph on state transition. Not sure whether this is doable (or how difficult) to do with the current AWS GSA.

One thing I haven't explored is whether this presence/absence of `publicIpAddress` in the instance API query response is correlated with whatever responses generate the graph linkages between instance and public gateway nodes (or whatever that latter node is called--can't remember off the top of my head).

**Comment by armaneous** [comment]
The signal would need to be empty string, since there's no concept of null in dgraph. And since it'll exist as a predicate `has(public_ip_address)` needs to instead check if the value of `public_ip_address` is _not_ `""`.

**Comment by armaneous** [comment]
*RESOLUTION*
Decision is to commit as is and track discussion to remember to resolve uid substitution in multiple places within the same query.

**Pull request merged**